main.go

```diff
@@ -35,6 +34,36 @@ type Entropy struct {
 	Line    string  // Line with high entropy
 }
 
+// Entropies should be created with a size n using make()
+// it should not be written to manually, instead use Entropies.Add
+type Entropies struct {
+	sync.Mutex
+	Entropies []Entropy
+}
+
+// Add assumes that es contains an ordered set of entropies.
+// It preserves ordering, and inserts an additional value e, if it has high enough entropy.
+// In that case, the entry with lowest entropy is rejected.
+func (es *Entropies) Add(e Entropy) {
+	es.Lock()
+	defer es.Unlock()
+
+	entropies := es.Entropies
+	if entropies[len(entropies)-1].Entropy >= e.Entropy {
+		return
+	}
+
+	for i := range len(entropies) {
+		if entropies[i].Entropy < e.Entropy {
+			for j := len(entropies) - 1; j > i; j-- {
+				entropies[j] = entropies[j-1]
+			}
+			entropies[i] = e
+			return
+		}
+	}
```

Comment:
```go
i, _ := slices.BinarySearchFunc(es.Entropies, e, func(a, b Entropy) int {
	return b.Entropy - a.Entropy
})

if i < len(es.Entropies) {
	copy(es.Entropies[i+1:], es.Entropies[i:])
	es.Entropies[i] = e
}
```

Reply:
Also
```go
func entropy(text string) float64 {
	unique := make(map[byte]float64, len(text))
	for i := 0; i < len(text); i++ { // count bytes, its faster than counting runes
		unique[text[i]] += 1.0
	}
	entropy := 0.0
	for _, count := range unique {
		res := count / float64(len(text)) // res can not be 0 because count and len are not zero
		entropy -= res * math.Log2(res)
	}
	return entropy
}
```

Reply:
Nice, thanks!

### binary search

I added a modified version of the first suggestion. I left in the initial check for `if entropies[len(entropies)-1].Entropy >= e.Entropy { ...` since in most cases, the majority of lines won't need to be added at all, and we can potentially skip unnecessary O(logN) operations with this check.

Also needed to modify the way float difference is returned as an int.

### entropy algorithm

looks very good, but IMO not suited for this MR